Puppet/Interpreter/Types.hs

```diff
@@ -187,12 +188,19 @@ data HieraQueryType
     , _mergeHashArray :: Bool
     } deriving (Show)
 
--- | The type of the Hiera API function.
-type HieraQueryFunc m = Container Text -- ^ All the variables that Hiera can interpolate, the top level ones being prefixed with ::
+-- | The type of the Hiera API function associated to one hiera file.
```

Comment:
There can actually be multiple files! I guess it's more like "a given hierarchy" ?